src/main/java/hu/elte/eserial/util/StringUtils.java

```diff
@@ -29,4 +29,53 @@ public static String lowercaseFirstLetter(String str) {
 
         return new String(characters);
     }
+
+    /**
+     * Escapes a string.
+     *
+     * @param str the string to be escaped
+     * @return the escaped version of {@code str}
+     */
+    public static String escape(String str) {
+        if (str.isEmpty()) {
+            return str;
+        }
+
+        StringBuilder builder = new StringBuilder();
+
+        char[] characters = str.toCharArray();
+
+        for (Character character : characters) {
```

Comment:
This solution uses manually assigns almost the same output to the input. This is prone to errors when copy pasting for example.

I'd prefer something like this:
1. Having an array of `\` escapable characters
1. Having another array of `\\` escapable characters
1. Checking if the input is in the first array -> return `\` + input
1. Checking if the input is in the first array -> return `\\` + input 

This minimizes code duplication, but is significantly slower.